jenkins/Jenkinsfile

```diff
@@ -437,22 +498,33 @@ timestamps {
                                     def builders = [:]
 
                                     // Add test runs on older macOS versions
-                                    def url = "https://bosstoragemirror.blob.core.windows.net/wrench/${virtualPath}/tests/mac-test-package.zip"
-                                    def firstOS = sh (returnStdout: true, script: "grep ^MIN_OSX_SDK_VERSION= '${workspace}/xamarin-macios/Make.config' | sed 's/.*=//'").trim ().split ("\\.")[1].toInteger ()
-                                    def lastOS = sh (returnStdout: true, script: "grep ^OSX_SDK_VERSION= '${workspace}/xamarin-macios/Make.config' | sed 's/.*=//'").trim ().split ("\\.")[1].toInteger ()
-                                    for (os = firstOS; os < lastOS; os++) {
-                                        def macOS = "${os}" // Need to bind the label variable before the closure
-                                        builders ["XM tests on 10.${macOS}"] = {
-                                            try {
-                                                node ("xamarin-macios && macos-10.${macOS}") {
-                                                    stage ("Running XM tests on '10.${macOS}'") {
-                                                        runXamarinMacTests (url, "macOS 10.${macOS}")
+                                    if (hasXamarinMacTests) {
+                                        def url = "https://bosstoragemirror.blob.core.windows.net/wrench/${virtualPath}/tests/mac-test-package.zip"
+                                        def maccore_hash = sh (script: "grep NEEDED_MACCORE_VERSION ${workspace}/xamarin-macios/mk/xamarin.mk | sed 's/NEEDED_MACCORE_VERSION := //'", returnStdout: true).trim ()
+                                        def firstOS = sh (returnStdout: true, script: "grep ^MIN_OSX_SDK_VERSION= '${workspace}/xamarin-macios/Make.config' | sed 's/.*=//'").trim ().split ("\\.")[1].toInteger ()
+                                        def lastOS = sh (returnStdout: true, script: "grep ^OSX_SDK_VERSION= '${workspace}/xamarin-macios/Make.config' | sed 's/.*=//'").trim ().split ("\\.")[1].toInteger ()
+                                        def macOSes = []
+                                        for (os = firstOS; os < lastOS; os++)
+                                            macOSes.add (os)
+                                        macOSes.add (14) // 10.14, Mojave
```

Comment:
> Can we reduce the sh a bit ? it's harder to read :)

I'm not sure how to make it simpler with how it's currently supposed to behave. Did you have anything in mind? Or would adding comments suffice?

> at some point we might want to disable (stop supporting) an old macOS version. It be easier to cherry-pick/backport changes to the script if the data (OS list) was clearly separated from the logic.

The list of macOS versions to execute tests on is derived from the MIN_OSX_SDK_VERSION value in Make.config, which means that updating the min osx version we supported means we'll automatically drop running tests on the macOS versions we drop support for.

The whole point of this chunk if code is to not have to update anything, we'd automatically run the XM tests on all (and only) the macOS versions we support.

Of course, if we want to poke holes in the assumption that we'd want to run XM tests on all macOS versions we support then the logic becomes more complicated :smile:

A couple of ideas:

* Have a complete list of the all the macOS versions (in a separate file for instance). Most flexible, but it's also effectively a duplication of data, that someone must remember to update. It's unlikely that it would need frequent updates (which has a downside: it means it'll be less likely that people will remember to update it).
* Compute the list list of macOS versions, and have a manually maintained black lists of macOS versions we don't want to run on (for the scenario yesterday, to skip 10.14 temporarily). A blacklist would also make it easier to show in Jenkins that a macOS version we support did not run the tests.

Reply:
A blacklist would do or, minimally something in the `README.md` that make it obvious (without reading thru the script) how do disable any macOS bot.